### ISSUE TITLE
State Changes not being picked up by protractor while using Phantomjs

### ISSUE THREAD

**Issue by ghost** [angular/protractor#986]
I am using protractor with PhantomJs. Simple tests seem to be working but when it comes to submitting a login form and matching a new url, protractor seems to be failing:
```js
describe("Login Page",function(){

  var ptor, logonID, password, signInButton;

  beforeEach(function(){
    ptor = protractor.getInstance();
    ptor.get('/login');
    ptor.findElement(protractor.By.id('logonID')).sendKeys('joe');
    password = element(by.model('user.password')).sendKeys('123');
    element(by.partialButtonText('Sign In')).click();
    ptor.waitForAngular(); // using this or keeping it out has not made any difference
  });
  
it("should re-direct user to another page once the user enters correct authentication information",function(){
      ptor = protractor.getInstance();
      ptor.getCurrentUrl().then(function(newUrl){
        expect(newUrl).toContain('dashboard');
      });
    });
 
});
```
Here is the error that protractor is throwing:

Error: Error while running testForAngular: Error Message => 'Detected a page unload event; asynchronous script execution does not work across page loads.'

A lot of other protractor tests in my test-suite are working, however I am not able to get the login form to work due to which i am not able to further test a huge chunk of my application (the part that is accessible once the user is logged in).

What am I missing here? Any clues ?

**Comment by senayar** [comment]
+1 same issue with phantomJS for me, he pass with Chrome. I tried with model and CSS input path.

```javascript
describe("Login Page", function () {

  beforeEach(function () {
    browser.get('http://localhost:9007/#/login');
    browser.driver.manage().window().setSize(1280, 1024);
  });

  it('should login user with correct login/password', function () {
    emailInput.clear();
    passwordInput.clear();
    //emailInput.sendKeys('test@user.fr');
    //passwordInput.sendKeys('passss');
    element(by.model('user.username')).sendKeys('test@user.fr');
    element(by.model('user.password')).sendKeys('passss');
    element(by.css('#sb-site > form > input.btn')).click().then(function() {
      browser.waitForAngular();
      expect(browser.getCurrentUrl()).toBe('http://localhost:9007/#/home');
    });
  });
});

**Comment by senayar** [comment]
it's look like is the same issue as : http://stackoverflow.com/questions/24431473/issues-with-using-phantomjs-and-protractor

**Comment by senayar** [comment]
I finally found how to fix it : I was using phantomJS 1.9.x and not the 2.x

So you need to uninstall phantomjs 1.9

npm remove -g phantomjs
and install phantomJS 2.x

npm install -g phantomjs2

and now the test work with PhantomJS

**Comment by juliemr** [comment]
Sounds like this is fixed - closing!
